libs/community/langchain_community/document_loaders/batch_loader.py

```diff
@@ -0,0 +1,288 @@
+    Using a class that is instantiate with __init__
+    ```
+    file_path_list : List[str] = [...]
+    loader_files = BatchLoader(
+        TextLoader,
+        {
+            "file_path": file_path_list
+        },
+        method="process",
+        max_workers=4,
+    )
+    ```
+    Using a static method to initialize the loader:
+    ```
+    youtube_video_urls : List[str] = [...]
+    loader_youtube = BatchLoader(
+        YoutubeLoader.from_url,
+        {
+            "url": youtube_video_urls
+        },
+    )
+    ```
+
+    Then you can load your data:
+    ```
+    documents = loader_files.load()
+    ```
+    """
+
+    def __init__(
+        self,
+        loader_initializer_callable: Callable[..., BaseLoader],
+        init_loader_args: Dict[str, List[Any]],
+        *,
+        method: Methods = "sequential",
+        max_workers: int = 1,
+        show_progress: bool = False,
+    ):
+        """Initialize the loader with a loader_callable (either a class or a
+        function used to initialize the loader) and a list of loader_args.
+
+        Possible methods are:
+            - "sequential": load data sequentially
+            - "thread": load data using threads
+            - "process": load data using processes
+            - "async": load data using asyncio
+
+        Args:
+            loader_initializer_callable (Callable[..., BaseLoader]): callable
+                used to initialize the loader (class or init function )
+            init_loader_args (Dict[str, List[Any]]): list of loader_args used
+                to initialize the loader. The keys of the dict are the name of
+                the arguments and the values are the list of values for each.
+            method (str, optional): concurrent method to use.
+                Defaults to "sequential".
+            max_workers (int, optional): number of workers if 'thread' or
+                'process' methods are used. Defaults to 1.
+            show_progress: If true, will show a progress bar as the files are
+                loaded. This forces an iteration through all matching files to
+                count them prior to loading them.
+
+        Raises:
+            ValueError: if the length of the lists in loader_args are not equal
+        """
+        loader_args = list(zip(*init_loader_args.values()))
+        self.loaders_args = [
+            dict(zip(init_loader_args.keys(), loader_arg)) for loader_arg in loader_args
+        ]
+        self.loader_callable = loader_initializer_callable
+        self.show_progress = show_progress
+        self._partial_load = partial(_load_loader, loader_callable=self.loader_callable)
+        self.method = method
+        self.max_workers = max_workers
+
+        # Check that all lists in loader_args have the same length
+        if len(set(len(arg) for arg in init_loader_args.values())) != 1:
+            raise ValueError("All loader_args lists must have the same length")
+
+    def load(self) -> List[Document]:
```

Comment:
`load` shouldn't be overridden for new loaders.
It's `lazy_load` that should be implemented.